### ISSUE TITLE
isLanguageAvailable returns false in release mode

### ISSUE THREAD

**Issue by alexandergottlieb** [dlutton/flutter_tts#73]
## 🐛 Bug Report

Hi, I'm testing my app on a physical Android device. 

In debug mode, `isLanguageAvailable('de-DE')` returns true but, in **release** mode, it returns false.

Calling `getLanguages` throws the below exception in release mode:

```
E/flutter (28498): [ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: PlatformException(error, Attempt to invoke interface method 'java.util.Iterator java.util.Set.iterator()' on a null object reference, null)
E/flutter (28498): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:569)
E/flutter (28498): #1      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:316)
E/flutter (28498): <asynchronous suspension>
E/flutter (28498): #2      FlutterTts.get:getLanguages (package:flutter_tts/flutter_tts.dart:55)
E/flutter (28498): <asynchronous suspension>
E/flutter (28498): #3      Bloc._initTextToSpeech (package:grammapp/bloc.dart:332)
E/flutter (28498): <asynchronous suspension>
E/flutter (28498): #4      new Bloc. (package:grammapp/bloc.dart:132)
E/flutter (28498): #5      _BlocProviderState.initState (package:grammapp/widgets/inherited_bloc.dart:37)
E/flutter (28498): #6      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:4068)
E/flutter (28498): #7      ComponentElement.mount (package:flutter/src/widgets/framework.dart:3919)
E/flutter (28498): #8      Element.inflateWidget (package:flutter/src/widgets/framework.dart:3101)
E/flutter (28498): #9      Element.updateChild (package:flutter/src/widgets/framework.dart:2904)
E/flutter (28498): #10     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3961)
E/flutter (28498): #11     Element.rebuild (package:flutter/src/widgets/framework.dart:3738)
E/flutter (28498): #12     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3924)
E/flutter (28498): #13     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3919)
E/flutter (28498): #14     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3101)
E/flutter (28498): #15     Element.updateChild (package:flutter/src/widgets/framework.dart:2904)
E/flutter (28498): #16     RenderObjectToWidgetElement._rebuild (package:flutter/src/widgets/binding.dart:998)
E/flutter (28498): #17     RenderObjectToWidgetElement.mount (package:flutter/src/widgets/binding.dart:969)
E/flutter (28498): #18     RenderObjectToWidgetAdapter.attachToRenderTree.<anonymous closure> (package:flutter/src/widgets/binding.dart:915)
E/flutter (28498): #19     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2328)
E/flutter (28498): #20     RenderObjectToWidgetAdapter.attachToRenderTree (package:flutter/src/widgets/binding.dart:914)
E/flutter (28498): #21     WidgetsBinding.attachRootWidget (package:flutter/src/widgets/binding.dart:795)
E/flutter (28498): #22     runApp (package:flutter/src/widgets/binding.dart:845)
E/flutter (28498): #23     main (package:grammapp/main.dart:16)
E/flutter (28498): #24     _runMainZoned.<anonymous closure>.<anonymous closure> (dart:ui/hooks.dart:229)
E/flutter (28498): #25     _rootRun (dart:async/zone.dart:1124)
E/flutter (28498): #26     _CustomZone.run (dart:async/zone.dart:1021)
E/flutter (28498): #27     _runZoned (dart:async/zone.dart:1516)
E/flutter (28498): #28     runZoned (dart:async/zone.dart:1500)
E/flutter (28498): #29     _runMainZoned.<anonymous closure> (dart:ui/hooks.dart:221)
E/flutter (28498): #30     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:305)
E/flutter (28498): #31     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172)
```

### Expected behavior

Language availability should not change in release mode.

### Reproduction steps

Run on an Android device in debug mode, then in release mode, calling `isLanguageAvailable('de-DE') both times`.

### Configuration

**Version:** 0.7.0

**Platform:** 
- [ ] :iphone: iOS
- [x] :robot: Android

**Comment by dlutton** [comment]
I have ran into this issue before and the way I resolved it was to use something like I provide in the example.

` 
if (Platform.isAndroid) {
      flutterTts.ttsInitHandler(() {
        _getLanguages();
      });
    }
`

I added this initHandler so that it won't call your getLanguages method until the android tts engine has been initialized.

**Comment by alexandergottlieb** [comment]
Perfect, that's fixed it!

I wonder if we could simplify the API a little? I've now got:

```dart
_textToSpeech = FlutterTts();
if (Platform.isAndroid) {
    final isAndroidInitialized = Completer();
    _textToSpeech.ttsInitHandler(() => isAndroidInitialised.complete());
    await isAndroidInitialized.future;
}
//Do platform-agnostic setup...
```

If that `Platform` check could be included somewhere in the library itself, the end developer would only have to write something like:

```dart
_textToSpeech = FlutterTts();
await _textToSpeech.isInitialized;
//Do platform-agnostic setup...
```

I can put this together in a pull request if you approve?

**Comment by dlutton** [comment]
sounds good @alexandergottlieb .  I always welcome PRs.

**Comment by dlutton** [comment]
@alexandergottlieb I have merged your PR.  Let me know if this resolves your issue so we can close this out.

**Comment by alexandergottlieb** [comment]
@dlutton thanks, it's working perfectly now. I'll go ahead and close this.

**Comment by alexandergottlieb** [comment]
Turns out the initial fix for this (using `CountDownLatch`) was occasionally causing ANRs on launch.

I've rewritten it to delay method calls made while the TTS engine is initialising using `Runnable` instead. The fix is in #81.

**Comment by dlutton** [comment]
The fix has been merged.  Thanks @alexandergottlieb
